### PR TITLE
scheduler: remove component config v1beta2

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
scheduler: remove deprecated component config v1beta2 since v1.26.

https://github.com/kubernetes/kubernetes/blob/9ec888e8737bf2b2132d2f2cb023b6c85fb857c9/cmd/kube-scheduler/app/options/configfile.go#L56

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
kube-scheduler: support for configuration file version `v1beta2` is removed. Update configuration files to v1beta3 or v1 before upgrading to 1.26.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
-[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/785-scheduler-component-config-api
```

### CODE REVIEW

**Comment by SataQiu** [comment]
/cc @kerthcet 
Not sure if we can remove v1beta2 now, or we should wait for some time.

**Comment by kerthcet** [comment]
> /cc @kerthcet 
> 
> Not sure if we can remove v1beta2 now, or we should wait for some time.

Yes, this is as planned,I'll take a look ASAP. Besides, let's update the KEP first.

**Comment by SataQiu** [comment]
> /retest

https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-local-e2e
It should be irrelevant to this PR, and it is not a Required test.

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -863,7 +863,7 @@ function start_kubescheduler {
     SCHEDULER_LOG=${LOG_DIR}/kube-scheduler.log
 
     cat <<EOF > /tmp/kube-scheduler.yaml
-apiVersion: kubescheduler.config.k8s.io/v1beta2
+apiVersion: kubescheduler.config.k8s.io/v1beta3
```
Comment (kerthcet) [inline hack/local-up-cluster.sh:866]: update to v1 seems better.

File: pkg/scheduler/apis/config/scheme/scheme_test.go [hunk 2]
```diff
@@ -42,347 +41,6 @@ func TestCodecsDecodePluginConfig(t *testing.T) {
 		wantErr      string
 		wantProfiles []config.KubeSchedulerProfile
 	}{
-		// v1beta2 tests
-		{
-			name: "v1beta2 all plugin args in default profile",
-			data: []byte(`
```
Comment (kerthcet) [inline pkg/scheduler/apis/config/scheme/scheme_test.go:48]: We have all the similar testcases in v1beta3, right?

File: test/integration/scheduler_perf/util.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ const (
 var dataItemsDir = flag.String("data-items-dir", "", "destination directory for storing generated data items for perf dashboard")
 
 func newDefaultComponentConfig() (*config.KubeSchedulerConfiguration, error) {
-	gvk := v1beta2.SchemeGroupVersion.WithKind("KubeSchedulerConfiguration")
+	gvk := v1beta3.SchemeGroupVersion.WithKind("KubeSchedulerConfiguration")
```
Comment (kerthcet) [inline test/integration/scheduler_perf/util.go:62]: maybe v1, it's stable now.

File: pkg/scheduler/apis/config/scheme/scheme_test.go (hunk 2 above)
Comment (SataQiu) [reply pkg/scheduler/apis/config/scheme/scheme_test.go:48]: Yes, we have.

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (SataQiu) [reply hack/local-up-cluster.sh]: Updated.

File: test/integration/scheduler_perf/util.go (hunk 3 above)
Comment (SataQiu) [reply test/integration/scheduler_perf/util.go]: Updated.

**Comment by SataQiu** [comment]
Wait for 9 months or 3 releases (whichever is longer) after deprecation

/hold

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by kerthcet** [comment]
Hi @SataQiu do you want to continue on this?

**Pull request closed without merging**